### PR TITLE
Removed version pinning for ransack dependency

### PR DESCRIPTION
**Pull request by petewalker**

### CODE REVIEW

**Comment by shioyama** [comment]
Hmm.. specs failing. I think they merged polyamorous into ransack, but not sure why Mobility would be affected. I'll need to look into it.

File: mobility-ransack.gemspec [hunk 1]
```diff
@@ -16,7 +16,7 @@ Gem::Specification.new do |spec|
   spec.files        = Dir['{lib/**/*,[A-Z]*}']
   spec.require_paths = ["lib"]
 
-  spec.add_dependency 'ransack', '>= 1.8.0', '< 2.1'
+  spec.add_dependency 'ransack', '>= 1.8.0'
```
Comment (shioyama) [inline mobility-ransack.gemspec]: Instead of making this a totally open-ended dependency, how about: `'>= 1.8.0', '< 3'` ?

**Comment by shioyama** [comment]
Maybe related: https://github.com/activerecord-hackery/ransack/issues/989

**Comment by shioyama** [comment]
Ah, you probably need to remove this from the `Gemfile` as well:

https://github.com/shioyama/mobility-ransack/blob/a57403619f3d7dd212ab0f7d1195f1caec438074/Gemfile#L15-L16

File: mobility-ransack.gemspec (hunk 1 above)
Comment (petewalker) [reply mobility-ransack.gemspec]: Done!

**Comment by petewalker** [comment]
The main problem seems to be a change in the join type:

```
expected: "SELECT \"posts\".* FROM \"posts\" INNER JOIN \"mobility_string_translations\" \"Post_title_en_string..._translations\".\"value\" LIKE '%foo%' AND \"Post_content_en_text_translations\".\"value\" = 'bar')"
            got: "SELECT \"posts\".* FROM \"posts\" LEFT OUTER JOIN \"mobility_string_translations\" \"Post_title_en_s..._translations\".\"value\" LIKE '%foo%' AND \"Post_content_en_text_translations\".\"value\" = 'bar')"
```

It seems to have changed from `INNER JOIN` to `LEFT OUTER JOIN`.

**Comment by shioyama** [comment]
> It seems to have changed from INNER JOIN to LEFT OUTER JOIN.

That's fine, can you just update the spec to match the `OUTER JOIN` ?

**Comment by petewalker** [comment]
Took a few commits - but seems to be passing now..!

**Comment by shioyama** [comment]
Thanks!

Just for the record (and for reference), the change from INNER to OUTER join is caused by shioyama/mobility#279, not related to anything in ransack. The `content_eq` predicate gets an `INNER` because it's an equality predicate with a non-nil value, but `title_cont` isn't explicitly handles and now falls through to `OUTER`, although `INNER` would also work (and would be faster). Maybe Mobility can improve on this in the future, but anyway not related to this gem.

**Pull request merged**